pkg/streamingpromql/engine_test.go

```diff
@@ -1923,7 +1922,7 @@ func TestCompareVariousMixedMetricsBinaryOperations(t *testing.T) {
 	expressions := []string{}
 
 	for _, labels := range labelCombinations {
-		for _, op := range []string{"+", "-", "*", "/", "and", "unless"} {
+		for _, op := range []string{"+", "-", "*", "/", "and", "unless", "or"} {
```

Comment:
It might be worth splitting `and`, `unless`, and `or` into their own `TestCompare...` function and then to also test them with some grouping.